libc/Makefile

```diff
@@ -17,6 +17,19 @@ LOCAL_LDFLAGS := -z stack-size=12288 $(2)
 include $(binary.mk)
 endef
 
+# Add libc test using custom sources from subdir
+# $(eval $(call add_libc_test,SUBDIR_NAME,NAME[,LOCAL_LDFLAGS][,LOCAL_CFLAGS]))
+define add_test_libc_custom
+NAME := test-libc-$(2)
+LIBC_UNIT_TESTS += test-libc-$(2)
+SRCS := $(addprefix $(MY_LOCAL_DIR)$(1)/, $(CUSTOM_SRCS)) $(COMMON_SRCS)
```

Comment:
- using external variable in macro instead of param is a way to introduce obscure errors
- IMHO you should either write it without a macro or specify a macro that just takes the list of c files to compile
- why not a single binary for socket tests?

Reply:
I create 2 binaries because `unix-sockets` test is run on all targets and `inet-sockets` test is run on targets with lwip
![image](https://github.com/phoenix-rtos/phoenix-rtos-tests/assets/108683715/26817a14-8536-4e97-988a-4afd8a3f477e)

Reply:
Still could be a single binary - with different test groups and using -g param in runtime to split the inet/unix tests - if it's too cumbersome to split them into different binaries (I assume all tests should compile regardless of the target architecture).

Side note: please don't add extra tests in this PR, we should aim for minimal changes that make them work and open new PR to bring new tests/functionality.